android/src/main/java/com/gettipsi/stripe/dialog/AddCardDialogFragmentTwo.java

```diff
@@ -0,0 +1,220 @@
+              ((SecurityCodeText) view).addTextChangedListener(new TextWatcher() {
+                @Override
+                public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                  Log.d(TAG, "beforeTextChanged: " + charSequence + " i = " + i + " i1 = " + i1 + " i2 = " + i2);
+                }
+
+                @Override
+                public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                  Log.d(TAG, "onTextChanged: " + charSequence + " i = " + i + " i1 = " + i1 + " i2 = " + i2);
+                  doneButton.setEnabled(charSequence.length() == 3);
+                }
+
+                @Override
+                public void afterTextChanged(Editable editable) {
+                  Log.d(TAG, "afterTextChanged: " + editable.toString());
+                }
+              });
+            }
+          } else {
+            Log.d(TAG, "ANIMATE: HIDE_CCV");
+            cardFlipAnimator.showFront();
+          }
+        }
+
+      }
+    });
+
+    cardFlipAnimator = new CardFlipAnimator(getActivity(), imageFlipedCard, imageFlipedCardBack);
+    succesful = false;
+  }
+
+  public void onSaveCLick() {
+    doneButton.setEnabled(false);
+    progressBar.setVisibility(View.VISIBLE);
+    final CreditCard fromCard = from.getCreditCard();
+    final Card card = new Card(
+      fromCard.getCardNumber(),
+      fromCard.getExpMonth(),
+      fromCard.getExpYear(),
+      fromCard.getSecurityCode());
+
+    String errorMessage = validateCard(card);
+    if (errorMessage == null) {
+      new Stripe().createToken(
+        card,
+        PUBLISHABLE_KEY,
+        new TokenCallback() {
+          public void onSuccess(Token token) {
+            final WritableMap newToken = Arguments.createMap();
+            newToken.putString("token", token.getId());
+            newToken.putBoolean("live_mode", token.getLivemode());
+            newToken.putBoolean("user", token.getUsed());
+            if (promise != null) {
+              promise.resolve(newToken);
+              promise = null;
+            }
+            succesful = true;
+            dismiss();
+          }
+
+          public void onError(Exception error) {
+            doneButton.setEnabled(true);
+            progressBar.setVisibility(View.GONE);
+            Toast.makeText(getActivity(), error.getLocalizedMessage(), Toast.LENGTH_LONG).show();
+          }
+        });
+    } else {
+      doneButton.setEnabled(true);
+      progressBar.setVisibility(View.GONE);
+      Toast.makeText(getActivity(), errorMessage, Toast.LENGTH_LONG).show();
+    }
+  }
+
+  private String validateCard(Card card) {
+    if (!card.validateNumber()) {
+      return "The card number that you entered is invalid";
+    } else if (!card.validateExpiryDate()) {
+      return "The expiration date that you entered is invalid";
+    } else if (!card.validateCVC()) {
+      return "The CVC code that you entered is invalid";
```

Comment:
I've seen similar code in previous file, please consider make proper superclass and reuse functions